core/nuxeo-connection.js

```diff
@@ -148,12 +154,16 @@ import './nuxeo-element.js';
         schemas: ['*'],
       };
 
-      if (this.username) {
-        options.auth = {
-          method: 'basic',
-          username: this.username,
-          password: this.password,
-        };
+      if (this.method === 'basic') {
+        if (this.username) {
```

Comment:
nit: nested if is harder to read. Inverting the if condition would reduce on leve of indentation:
```
if (!this.method === 'basic') {
  options.headers = { 'X-No-Basic-Header': true };
} else if (this.username) {
  options.auth = {
    method: 'basic',
    username: this.username,
    password: this.password,
  };
}
```